plugins/Phaser/PhaserEffect.cpp

```diff
@@ -0,0 +1,623 @@
+			{
+				tempAdd[0] = delayOut[0] * feedback;
+				tempAdd[1] = delayOut[1] * feedback;
+				break;
+			}
+			case 2:
+			{
+				tempAdd[0] = lastSecondAdd[0] * feedback;
+				tempAdd[1] = lastSecondAdd[1] * feedback;
+				break;
+			}
+			case 3:
+			{
+				tempAdd[0] = delayOut[0] * feedback;
+				tempAdd[1] = delayOut[1] * feedback;
+				for (int i = 0; i < order; ++i)
+				{
+					for (int j = 0; j < 2; ++j)
+					{
+						for (int k = 0; k < 1 + doubleVal; ++k)// Runs twice if doubleVal is true
+						{
+							tempAdd[j] = calcAllpassFilter(tempAdd[j], sample_rate, i, j, b0[j], b1[j]);
+						}
+					}
+				}
+			}
+		}
+		if (analog)
+		{
+			firstAdd[0] = tanh(s[0] * analogDist) / analogDist + tanh(tempAdd[0] * analogDist) / analogDist;
+			firstAdd[1] = tanh(s[1] * analogDist) / analogDist + tanh(tempAdd[1] * analogDist) / analogDist;
+		}
+		else
+		{
+			firstAdd[0] = s[0] + tempAdd[0];
+			firstAdd[1] = s[1] + tempAdd[1];
+		}
+		
+
+		
+		// Unleash the allpass filters
+		switch (mode)
+		{
+			case 0: case 1: case 3:
+			{
+				secondAdd[0] = firstAdd[0];
+				secondAdd[1] = firstAdd[1];
+				break;
+			}
+			case 2:
+			{
+				secondAdd[0] = delayOut[0];
+				secondAdd[1] = delayOut[1];
+				break;
+			}
+		}
+
+		if (alias)
+		{
+			secondAdd[0] *= m_aliasFlip;
+			secondAdd[1] *= m_aliasFlip;
+		}
+		for (int i = 0; i < order; ++i)
+		{
+			for (int j = 0; j < 2; ++j)
+			{
+				for (int k = 0; k < 1 + doubleVal; ++k)// Runs twice if doubleVal is true
+				{
+					secondAdd[j] = calcAllpassFilter(secondAdd[j], sample_rate, i, j, b0[j], b1[j]);
+				}
+			}
+		}
+		if (alias)
+		{
+			secondAdd[0] *= m_aliasFlip;
+			secondAdd[1] *= m_aliasFlip;
+		}
+
+		// Get final result
+		switch (mode)
+		{
+			case 0: case 3:
+			{
+				s[0] = (secondAdd[0] * (invert ? -1 : 1) + firstAdd[0] * !wetIsolate) / 2.f;
+				s[1] = (secondAdd[1] * (invert ? -1 : 1) + firstAdd[1] * !wetIsolate) / 2.f;
+				break;
+			}
+			case 1:
+			{
+				s[0] = (secondAdd[0] * (invert ? -1 : 1) + s[0] * !wetIsolate) / 2.f;
+				s[1] = (secondAdd[1] * (invert ? -1 : 1) + s[1] * !wetIsolate) / 2.f;
+				break;
+			}
+			case 2:
+			{
+				s[0] = (secondAdd[0] * (invert ? -1 : 1) + firstAdd[0] * !wetIsolate * -feedback) / 2.f;
+				s[1] = (secondAdd[1] * (invert ? -1 : 1) + firstAdd[1] * !wetIsolate * -feedback) / 2.f;
+				break;
+			}
+		}
+
+		lastSecondAdd[0] = secondAdd[0];
+		lastSecondAdd[1] = secondAdd[1];
+
+		tempAdd[0] = tempAdd[1] = 0;
+		switch (mode)
+		{
+			case 0: case 1: case 3:
+			{
+				tempAdd[0] = secondAdd[0];
+				tempAdd[1] = secondAdd[1];
+				break;
+			}
+			case 2:
+			{
+				tempAdd[0] = firstAdd[0];
+				tempAdd[1] = firstAdd[1];
+				break;
+			}
+		}
+
+		/*
+		Some gentle feedback distortion to prevent infinite loops.
+
+		I used tanh(x/2)*2 rather than just tanh(x) because tanh(x)
+		decreased the volume too much for my liking.
+		*/
+		tempAdd[0] = tanh(tempAdd[0] * 0.5) * 2;
+		tempAdd[1] = tanh(tempAdd[1] * 0.5) * 2;
```

Comment:
Has the aliasing been considered? although I feel this``` tanh``` needs to remain, as it is much preferable to the infinite signal rise that could happen if removed